core/src/date.ts

```diff
@@ -7,6 +7,9 @@ const CURRENT_BLOCK_DATE_CACHE_EXPIRY_MS = 60 * 1000;
 export const fetchDateByBlockNumber = async function (network: string, blockNumber: number): Promise<Date> {
     const provider = await getProvider(network);
     const block = await provider.getBlock(blockNumber);
+    if (!block?.timestamp) {
+        throw new Error('Received incorrect timestamp value');
+    }
```

Comment:
Last question: what happens one the frontend if this function throws an error? Where is it catched or does it crash the whole frontend? 🙂

Reply:
- in the auction store the call that fetches all aucitons (and has this in the call tree) is in try-except block.
- in the surplus store the call that fetches all surplus auctions also is in trycatch block
- no other calls that impact frontend were found by me.

Reply:
Given the previous answer it seems like we are handling the error properly. 
@valiafetisov if fine for you as well I would proceed and merge this one.